packages/next/src/build/webpack/plugins/flight-manifest-plugin.ts

```diff
@@ -334,94 +335,75 @@ export class ClientReferenceManifestPlugin {
-        // Remove parallel route labels:
-        // - app/foo/@bar/page -> app/foo
-        // - app/foo/@bar/layout -> app/foo/layout -> app/foo
-        const entryNameWithoutNamedSegments = entryName.replace(/\/@[^/]+/g, '')
-        const groupNameWithoutNamedSegments =
-          entryNameWithoutNamedSegments.slice(
-            0,
-            entryNameWithoutNamedSegments.lastIndexOf('/')
-          )
-        if (!manifestsPerGroup.has(groupNameWithoutNamedSegments)) {
-          manifestsPerGroup.set(groupNameWithoutNamedSegments, [])
-        }
-        manifestsPerGroup.get(groupNameWithoutNamedSegments)!.push(manifest)
-      }
+    // console.log(manifestEntryFiles, manifestsPerGroup)
 
-      // Special case for the root not-found page.
-      if (/^app\/not-found(\.[^.]+)?$/.test(entryName)) {
-        if (!manifestsPerGroup.has('app/not-found')) {
-          manifestsPerGroup.set('app/not-found', [])
-        }
-        manifestsPerGroup.get('app/not-found')!.push(manifest)
+    // Generate per-page manifests.
+    for (const pageName of manifestEntryFiles) {
+      const mergedManifest: ClientReferenceManifest = {
+        ssrModuleMapping: {},
+        edgeSSRModuleMapping: {},
+        clientModules: {},
+        entryCSSFiles: {},
       }
-    })
 
-    // Generate per-page manifests.
-    for (const [groupName] of manifestsPerGroup) {
-      if (groupName.endsWith('/page') || groupName === 'app/not-found') {
-        const mergedManifest: ClientReferenceManifest = {
-          ssrModuleMapping: {},
-          edgeSSRModuleMapping: {},
-          clientModules: {},
-          entryCSSFiles: {},
-        }
+      const segments = pageName.split('/')
+      let group = ''
+      for (const segment of segments) {
+        if (segment.startsWith('@')) continue
+        if (segment.startsWith('(') && segment.endsWith(')')) continue
 
-        const segments = groupName.split('/')
-        let group = ''
-        for (const segment of segments) {
-          if (segment.startsWith('@')) continue
-          for (const manifest of manifestsPerGroup.get(group) || []) {
-            mergeManifest(mergedManifest, manifest)
-          }
-          group += (group ? '/' : '') + segment
-        }
-        for (const manifest of manifestsPerGroup.get(groupName) || []) {
+        for (const manifest of manifestsPerGroup.get(group) || []) {
           mergeManifest(mergedManifest, manifest)
         }
+        group += (group ? '/' : '') + segment
+      }
 
-        const json = JSON.stringify(mergedManifest)
+      const json = JSON.stringify(mergedManifest)
 
-        const pagePath = groupName.replace(/%5F/g, '_')
-        const pageBundlePath = normalizePagePath(pagePath.slice('app'.length))
+      const pagePath = pageName.replace(/%5F/g, '_')
+      const pageBundlePath = normalizePagePath(pagePath.slice('app'.length))
+      assets[
+        'server/app' + pageBundlePath + '_' + CLIENT_REFERENCE_MANIFEST + '.js'
```

Comment:
nice to have later: can format the `CLIENT_REFERENCE_MANIFEST` with readable indent in development mode

Reply:
Not very sure about this one. Since the new manifest is no longer a JSON, but a JS file with the data stored as a JS string. In large applications I've observed client reference manifest growing too large so I think it's better to keep it minimized and performant (memory) by default.